Install.lua

```diff
@@ -1,175 +1,275 @@
+								Generated = true
+							end
+							Parent = Object
+						end
+					end
+				end
+			end
+		end
+		
+		Link:gsub("[^/]+/?$", ""):gsub("[^/]+", LocateFolder)
+
+		if MainExists or ScriptCount ~= 1 or ScriptName ~= LastFolder then
+			LocateFolder(LastFolder)
+		end
+		
+		local Script = Instance.new(ScriptTypes[ScriptClass or TypesSpecified and "" or "mod"], Parent)
+		Script.Name = ScriptName
+		if not Routines[1] then Routines[1] = Script end
+		coroutine.resume(coroutine.create(GiveSourceToScript), "https://raw.githubusercontent.com" .. ScriptLink:gsub("(/[^/]+/[^/]+/)blob/", "%1", 1), Script)
+		
+		if MainExists then
+			Parent = Script
+		end
 
-local MainDirectory = GetDirectory(ServerScriptService, "Nevermore\\")
-local Paths = {}
-for ScriptPath in (LoadURL("Modules/ModuleList.txt")):gmatch("[^\r\n]+") do
-	Paths[#Paths+1] = ScriptPath
-	print""
+		for a = 2, ScriptCount do
+			local Link = Scripts[a]
+			local ScriptName, ScriptClass = Link:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			local Script = Instance.new(ScriptTypes[ScriptClass or TypesSpecified and "" or "mod"], Parent)
+			Script.Name = ScriptName:gsub("Library$", "", 1):gsub("%%(%x%x)", UrlDecode)
+			coroutine.resume(coroutine.create(GiveSourceToScript), "https://raw.githubusercontent.com" .. Link:gsub("(/[^/]+/[^/]+/)blob/", "%1", 1), Script)
+		end
+	end
+
+	for a = 1, FolderCount do
+		local Link = Folders[a]
+		coroutine.resume(coroutine.create(InstallRepo), "https://github.com" .. Link, Directory, Parent, Routines, TypesSpecified)
+	end
+
+	Routines[Value] = true
 end
 
-for Index, ScriptPath in pairs(Paths) do
-	MakeScript(GetDirectory(MainDirectory, ScriptPath), "Modules/" .. ScriptPath)
-	ProgressText.Text = Index.."/"..#Paths
+function GitHub:Install(Link, Parent, RoutineList)
+	-- Installs Link into Parent
+	Link = Link:gsub("/$", "")
+
+	-- Extract Link Data
+	local Organization, Repository, Tree, ScriptName, ScriptClass
+	local Website, Directory = Link:match("^(https://[raw%.]*github[usercontent]*%.com/)(.+)")
+	Organization, Directory = (Directory or Link):match("^/?([%w-_%.]+)/?(.*)")
+	Repository, Directory = Directory:match("^([%w-_%.]+)/?(.*)")
+
+	if Website == "https://raw.githubusercontent.com/" then
+		if Directory then
+			Tree, Directory = Directory:match("^([^/]+)/(.+)")
+			if Directory then
+				ScriptName, ScriptClass = Directory:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			end
+		end		
+	elseif Directory then
+		local a, b = Directory:find("^[tb][rl][eo][eb]/[^/]+")
+		if a and b then
+			Tree, Directory = Directory:sub(6, b), Directory:sub(b + 1)
+			if Directory == "" then Directory = nil end
+			if Directory and Link:find("blob") then
+				ScriptName, ScriptClass = Directory:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			end
+		else
+			Directory = nil
+		end
+	end
+
+	if not Website then Website = "https://github.com/" end
+	Directory = Directory and ("/" .. Directory):gsub("^//", "/") or ""
+
+	-- Threads
+	local Routines = RoutineList or {false}
```

Comment:
Method shouldn't do both extraction and task scheduling like this, interloping with global variables

Reply:
Instead: split up into multiple variables/methods